algoliasearch/check_synonym.go

```diff
@@ -0,0 +1,27 @@
+package algoliasearch
+
+import "fmt"
+
+func checkSynonym(s Synonym) error {
+	switch s := s.(type) {
+	case AltCorrectionSynonym,
```

Comment:
Using an enum here is not possible, because some composed stuff is possible like : `altCorrection1|altCorrection2`